source/data_structures.gml

```
@@ -138,6 +138,197 @@
+    //list: ds list index
+    //lists: list(s) to concat
+    //Concats lists into the list
+    var __list, __currentList, __i, __j;
+    __list = argument0
+    for (__i = 1; __i < argument_count; __i += 1) {
+        __currentList = argument[__i]
+        for (__j = 0; __j < ds_list_size(__currentList); __j += 1) {
+            ds_list_add(__list, ds_list_find_value(__currentList, __j))
+        }
+    }
+
+
+#define ds_list_contains
+    ///ds_list_contains(list, value)
+    //list: ds list index
+    //value: value to check
+    //returns: bool
+    //Checks if the given value exists in the list.
+    return ds_list_find_index(argument0, argument1) != -1
+
+
+#define ds_list_contains_ext
+    ///ds_list_contains_ext(list, values...)
+    //list: ds list index
+    //values: values to check
+    //returns: bool
+    //Checks if the given values exists in the list.
+    var __array, __found, __i, __j;
+    __array = argument0
+    for (__i = 1; __i < argument_count; __i += 1) {
+        __found = false;
+        
+        for (__j = 0; __j < ds_list_size(__array); __j += 1) {
+            if (ds_list_find_value(__array, __j) == argument[__i]) {
+                __found = true
+                break;
+            }
+        }
+        
+        if (!__found) return false
+    }
+    return true
+
+
+#define ds_list_filter
+    ///ds_list_filter(list, callback)
+    //list: ds list index
+    //callback: callback script for all values
+    //Returns a new list that is the filtered version of the given list
+    var __list, __callback, __new, __i, __value;
+    __list = argument0
+    __callback = argument1
+    __new = ds_list_create()
+    if (script_exists(__callback)) {
+        for (__i = 0; __i < ds_list_size(__list); __i += 1) {
+            __value = ds_list_find_value(__list, __i);
+            if (script_execute(__callback, __value, __i)) {
+                ds_list_add(__new, __value)
+            }
+        }
+    }
+    return __new

```

this is going to return an empty list on arg1 failure, when it should probably throw an error and return undefined - is this really how it works in gml v2?

---

you mean if the script doesnt exist?

---

yeah, if the script doesn't exist it would be an error.

---

maybe i should also call show_error and return undefined

---

yeah, that would be ideal.